### PR TITLE
TCM-2256 Attachments API public docs and SDK

### PR DESCRIPTION
**Pull request by AlexanderNovak**

### CODE REVIEW

**Review by DmitryBochorishvili — COMMENTED** [review]
api spec looks good to me, however I see apparent grammatical mistakes in the description. Someone needs to proofread it.
Also please make sure all the sentences end with period.

File: spec/openapi.yaml [hunk 1]
```diff
@@ -325,6 +325,19 @@ tags:
       platform and specify additional filtering criteria to narrow down data
       returned by the reports.
 
+  - name: Attachments
+    description: |-
+      The Attachments API allows upload, download and link binary data with entity.
+      Unlinked attachments will be deleted after 2 days of the last upload
+      Client is responsible for entity uid validation and removing attachments within entity removal process
```
Comment (rachelsmartling) [inline spec/openapi.yaml:332]:
```suggestion
      The user is responsible for entity uid validation and removing attachments within the entity removal process
```

File: spec/openapi.yaml [hunk 2]
```diff
@@ -325,6 +325,19 @@ tags:
...
+      The following attachment types are supported:
+      - **jobs**: attachements for jobs
+      - **issues**: attachments for issues
+      - **strings**: attachments for strings
+
+      **Attachments API Usage Limitations**:
+      * Maximum file upload size is 500MB.
+
   - name: Translation quality checks
     description: The translation quality check service is an API endpoint at which you can check the quality of your translations.
```
Comment (rachelsmartling) [inline spec/openapi.yaml:342]: Is this related to configuring quality check profiles? If so, suggested edit: "The translation quality check service is an API endpoint at which you can set quality check profiles for linguists to following when translating your content in Smartling."

File: spec/openapi.yaml [hunk 3]
```diff
@@ -9722,6 +9735,352 @@ paths:
...
+            schema:
+              properties:
+                file:
+                  type: string
+                  format: binary
+                  description: |-
+                    The attachemnt file to upload. Maximum file upload size is `500MB`.
+                name:
+                  description: |-
+                    Functional name of the file uploaded. This is not a unique identifier.
```
Comment (rachelsmartling) [inline spec/openapi.yaml:9781]: Is there a character limit to the description?

File: spec/openapi.yaml [hunk 4]
```diff
@@ -9722,6 +9735,352 @@ paths:
...
+                  type: string
+                description:
+                  description: |-
+                    (Optional) Additional meta information of uploaded binary file
+                  type: string
+                entityUids:
+                  description: |-
+                    (Optional) List of entity uids attachment will be linked with (max 100)
```
Comment (rachelsmartling) [inline spec/openapi.yaml:9789]: (max 100) is max 100 uids or max 100 characters?

File: spec/openapi.yaml [hunk 5]
```diff
@@ -9722,6 +9735,352 @@ paths:
...
+          $ref: '#/components/responses/Error429ResponseDefinition'
+        500:
+          $ref: '#/components/responses/Error500ResponseDefinition'
+
+  '/attachments-api/v2/accounts/{accountUid}/{type}':
+    put:
+      summary: Link attachment with entity
+      description: |-
+        Link all attachments with all specified entity uids (cross join).
+        Max amount of uids are depend on attachment type and amount of already linked attachments per entity
```
Comment (rachelsmartling) [inline spec/openapi.yaml:9875]:
```suggestion
        Max amount of uids depend on the attachment type and the amount of existing linked attachments per entity
```

File: spec/openapi.yaml [hunk 6]
```diff
@@ -10125,6 +10484,92 @@ components:
...
+            - properties:
+                data:
+                  properties:
+                    items:
+                      description: Source terms
+                      items:
+                        $ref: '#/components/schemas/AttachmentResponseItem'
+                      type: array
+                    totalCount:
+                      description: total number of attachemnts
```
Comment (rachelsmartling) [inline spec/openapi.yaml:10515]: any limit?

**Review by rachelsmartling — CHANGES_REQUESTED** [review]
Please ensure any limits or restrictions are also included in the documentation.

File: spec/openapi.yaml [hunk 7]
```diff
@@ -325,6 +325,19 @@ tags:
...
 
+  - name: Attachments
+    description: |-
+      The Attachments API allows upload, download and link binary data with entity.
+      Unlinked attachments will be deleted after 2 days of the last upload
+      Client is responsible for entity uid validation and removing attachments within entity removal process
+      The following attachment types are supported:
+      - **jobs**: attachements for jobs
+      - **issues**: attachments for issues
+      - **strings**: attachments for strings
```
Comment (klegins-smartling) [inline spec/openapi.yaml:336]: I think we should add some short detaisl on how entity ids are constructed for the various domains?
@AlexArcher and @andreyfurdylo can you confirm those formulas for strings, jobs and issues?

File: spec/openapi.yaml (hunk 2 above)
Comment (AlexanderNovak) [reply spec/openapi.yaml:342]: no, this is just surrounding next to modified code. it's untouched.

File: spec/openapi.yaml (hunk 6 above)
Comment (AlexanderNovak) [reply spec/openapi.yaml:10515]: no paging there. it just duplicates number of array items

File: spec/openapi.yaml (hunk 7 above)
Comment (andreyfurdylo) [reply spec/openapi.yaml:336]: job uid - 12 random alpha-numeric (a-z0-9) characters.

Comment (AlexArcher) [reply spec/openapi.yaml:336]: for issue: {accountUid}-{issueUid}
for issue comment: {accountUid}-{issueUid}-{commentUid}
Examples:
8f14e45fc-0b0535d1d34a
8f14e45fc-0b0535d1d34a-c82010dd3bca

**Review by klegins-smartling — APPROVED** [review]

**Review by rachelsmartling — APPROVED** [review]
